src/app/lib/access/logger/logger_repository.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// Contract repository to handle logging configuration.
+abstract interface class LoggerRepository {
+  factory LoggerRepository() = _LoggerRepository;
+
+  /// Loads the current logging configuration.
+  Future<LoggingConfiguration> getLoggingConfiguration();
+
+  /// Sets whether console logging should be enabled on next app launch.
+  Future<bool> setIsConsoleLoggingEnabled(bool isConsoleLoggingEnabled);
+
+  /// Sets whether file logging should be enabled on next app launch.
+  Future<bool> setIsFileLoggingEnabled(bool isFileLoggingEnabled);
+}
+
+/// Entity that represents the current state of the logging configuration persisted.
+final class LoggingConfiguration {
```

Comment:
Was wondering if we should move it to a new file. Depends what the others think.

Reply:
It has been moved. I've also slightly changed the name of the file